enterprise/core-edge/src/main/java/org/neo4j/commandline/dbms/UnbindFromClusterCommand.java

```diff
@@ -0,0 +1,184 @@
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.fs.FileUtils;
+import org.neo4j.kernel.StoreLockException;
+import org.neo4j.kernel.internal.StoreLocker;
+
+import static java.lang.String.format;
+import static java.nio.file.Files.exists;
+import static org.neo4j.coreedge.core.EnterpriseCoreEditionModule.CLUSTER_STATE_DIRECTORY_NAME;
+
+public class UnbindFromClusterCommand implements AdminCommand
+{
+    public static class Provider extends AdminCommand.Provider
+    {
+        public Provider()
+        {
+            super( "unbind" );
+        }
+
+        @Override
+        public Optional<String> arguments()
+        {
+            return Optional.of( "[--database=<name>] " );
+        }
+
+        @Override
+        public String description()
+        {
+            return "Removes cluster state data from the specified database making it suitable for use non-Core-Edge " +
+                    "databases";
+        }
+
+        @Override
+        public AdminCommand create( Path homeDir, Path configDir, OutsideWorld outsideWorld )
+        {
+            return new UnbindFromClusterCommand( homeDir, new DefaultFileSystemAbstraction() );
+        }
+    }
+
+    private Path homeDir;
+    private FileSystemAbstraction fsa;
+
+    UnbindFromClusterCommand( Path homeDir, FileSystemAbstraction fsa )
+    {
+        this.homeDir = homeDir;
+        this.fsa = fsa;
+    }
+
+    @Override
+    public void execute( String[] args ) throws IncorrectUsage, CommandFailed
+    {
+        Args parsedArgs = Args.parse( args );
+
+        try
+        {
+            Path pathToSpecificDatabase =
+                    homeDir.resolve( "data" ).resolve( "databases" ).resolve( databaseNameFrom( parsedArgs ) );
+            confirmTargetDatabaseDirectoryExists( pathToSpecificDatabase );
+            confirmClusterStateDirectoryExists( Paths.get( pathToSpecificDatabase.toString(), "cluster-state" ) );
+            confirmTargetDirectoryIsWritable( pathToSpecificDatabase );
+            deleteClusterStateIn( clusterStateFrom( pathToSpecificDatabase ) );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            throw new IncorrectUsage( e.getMessage() );
+        }
+        catch ( UnbindFailureException e )
+        {
+            throw new CommandFailed( "Unbind failed: " + e.getMessage(), e );
+        }
+    }
+
+    private Path clusterStateFrom( Path target )
+    {
+        return Paths.get( target.toString(), CLUSTER_STATE_DIRECTORY_NAME );
+    }
+
+    private void confirmTargetDirectoryIsWritable( Path dbDir ) throws CommandFailed
+    {
+        Path lockFile = dbDir.resolve( StoreLocker.STORE_LOCK_FILENAME );
+        if ( exists( lockFile ) )
+        {
+            if ( Files.isWritable( lockFile ) )
+            {
+                StoreLocker storeLocker = new StoreLocker( fsa );
+                try
+                {
+                    storeLocker.checkLock( dbDir.toFile() );
```

Comment:
You don't unlock anywhere. I _think_ that's going to be okay with the current implementation because we're sure that after this command is done, the JVM will just exit. But since this class doesn't have a main method and is instead invoked by code from elsewhere, I don't think it's reasonable for the implementation to rely on an immediate exit. You may also run into problems with test ordering.